gandalf.js

```diff
@@ -7,15 +7,15 @@
  * See README.md for details.
```

Comment:
SCRIPT-The way you've refactored this complex method into simpler ones is great. It makes the code much more maintainable.